### DISCUSSION TITLE
about betadisper() interpretation

### DISCUSSION THREAD

**Discussion by Gian77** [vegandevs/vegan#483] [Q&A]
Hello, 

I I am using the function `betadisper()` after `adonis()` to test for homogeneity of variances. I have used it a lot in the years but I never get to look at the boxplots output with the necessary attention.

So this is my code. 

```
permdisp <- betadisper(vegdist(otu, method="bray"), metadata$Origin)
par(mfrow=c(1,2))
plot(permdisp, main="PCoA")
boxplot(permdisp, main="Distance to centroids")
```

![test](https://user-images.githubusercontent.com/13815699/106621132-3ff0f380-6540-11eb-8867-530a2c589856.png)

In the ordination plot looks like the Root samples should have smaller distances to centroids respect to the Soil samples. However, this is not what we see int he boxplot. is it because the PCoA is in 2D but in reality there are many axes, so the distances to the centroids might be bigger than what seen in PC1 & PC2. 

So I tried to plot the distance manually and I got this

`boxplot(permdisp$distances, permdisp$group)`

![test2](https://user-images.githubusercontent.com/13815699/106622433-8f83ef00-6541-11eb-81a8-a9ce7e47aeb6.png)

`SAM110 SAM111  SAM37 SAM112 SAM113 SAM114 SAM115 SAM116 SAM117 SAM118 SAM119 SAM120 
  Root   Root   Root   Soil   Soil   Soil   Soil   Root   Root   Root   Root   Soil 
Levels: Root Soil`

Levels 1 and 2 correspond exactly to Root and Soil. This plot makes more sense to me in relation to the PCoA graph. 
So, I am wondering if: 
* does the default function `boxplot()` on the betadisper object (as showed above) calculates distance distribution across all axis?
* does the `boxplot()` plotted manually actually show the distances seen in the 2D PCoA?
* Does then anova [`anova(permdisp, permutations = 9999)`] used tests for differences across group centroids using all PCoA axis or just the first 2?

Thanks you very much,

Gian

**Comment by jarioksa** [comment]
Your Box plot is wrongly constructed (or at least, it is unlikely to show what you want it to show). With this command
```r
boxplot(permdisp$distances, permdisp$group)
```
the plot will show two variables: the ungrouped distances (left) and the group values of 1 and 2 (right). If you want to show the distances *by* groups, you should use

```r
boxplot(permdisp$distances ~ permdisp$group)
```
This latter version is constructed by `vegan:::boxplot.betadisper`.

The PCoA plot shows only two first axes, but **all** calculations are based on all axes (or actually, in the original full space). So the `boxplot` is not based on only two axes, but on all. The `distances` returned by `betadisper` are also calculated in the full space over all dimensions, and `vegan:::anova.betadisper` uses those distances in its analysis.

**Comment by Gian77** [comment]
Hello @jarioksa,

thanks so much for your answer. it makes a lot of sense, and I could not find it in the `betadisper` help.

I think it is just kind of weird, for example in my case, that the PCoA shows a pattern of dispersion (based on 2 axis, 2D) that is opposite to what the all PCoA axes (original full space) is showing. 

Thanks,

G.
